File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    values = ['A', 'E', 'I', 'O', 'U']
    wordDict = []
    
    def dfs(cur):
        wordDict.append(cur)
        if len(cur) == 5: 
            return
        for i in range(5):
            dfs(cur + values[i])
        
    dfs('')
    return wordDict.index(word)

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    const values = ['A', 'E', 'I', 'O', 'U'];
    let wordDict = [];
    
    function dfs(cur){
        wordDict.push(cur);
        if(cur.length === 5) return;
        
        for(let i = 0 ; i < 5 ; i++){
            dfs(cur + values[i]);
        }
        
    }
    
    dfs('');
    return (wordDict.indexOf(word));
}